### ISSUE TITLE
Installation collision with libzip

### ISSUE THREAD

**Issue by reanimus** [javer/gentoo-overlay#22]
As noted in javer/gentoo-overlay#16, even while depending on the system libzip, hhvm builds its own and attempts to install it. This causes a package collision and thus an installation failure. It looks like someone there has provided a fix as well. Could it be merged in?

```
>>> Installing (1 of 1) dev-php/hhvm-3.2.0::javer
 * checking 1323 files for package collisions
1000 files checked ...
 * This package will overwrite one or more files that may belong to other
 * packages (see list below). You can use a command such as `portageq
 * owners / <filename>` to identify the installed package that owns a
 * file. If portageq reports that only one package owns a file then do
 * NOT file a bug report. A bug report is only useful if it identifies at
 * least two or more packages that are known to install the same file(s).
 * If a collision occurs and you can not explain where the file came from
 * then you should simply ignore the collision since there is not enough
 * information to determine if a real problem exists. Please do NOT file
 * a bug report at http://bugs.gentoo.org unless you report exactly which
 * two packages install the same file(s). See
 * http://wiki.gentoo.org/wiki/Knowledge_Base:Blockers for tips on how to
 * solve the problem. And once again, please do NOT file a bug report
 * unless you have completely understood the above message.
 *
 * Detected file collision(s):
 *
 *      /usr/lib/libzip.so
 *      /usr/include/zipconf.h
 *      /usr/include/zip.h
 *
 * Searching all installed packages for file collisions...
 *
 * Press Ctrl-C to Stop
 *
 * dev-libs/libzip-0.11.2:0::gentoo
 *      /usr/include/zip.h
 *      /usr/include/zipconf.h
 *      /usr/lib/libzip.so
 *
 * Package 'dev-php/hhvm-3.2.0' NOT merged due to file collisions. If
 * necessary, refer to your elog messages for the whole content of the
 * above message.
```

**Comment by lutel** [comment]
Why is it still not fixed? The ebuild is outdated :(

**Comment by reanimus** [comment]
@lutel i'm working on getting 3.5.0 working on my forked version of this overlay, if you're interested.

**Comment by lutel** [comment]
@reanimus thanks but i dont see it in layman list, how can I subscribe to your repository in gentoo?

**Comment by reanimus** [comment]
My repo isn't in the official overlay list. I created a bug [here](https://bugs.gentoo.org/show_bug.cgi?id=538974) to get it added.

In the meantime, you can use the instructions [here](http://blog.samuelololol.org/2010/10/layman-with-custom-git-repo-ie-github.html) to use a local list of overlays and add my github repo https://github.com/reanimus/hhvm-overlay to it.

**Comment by lutel** [comment]
@reanimus thank you, however there are some issues with the ebuilds:

 * ERROR: dev-php/hhvm-3.3.3::hhvm failed (prepare phase):
 *   Failed Patch: hhvm_use_system_libzip_libpcre.patch!
 *
 * Call stack:
 *     ebuild.sh, line   93:  Called src_prepare
 *   environment, line 2693:  Called epatch '/var/lib/layman/hhvm/dev-php/hhvm/files/hhvm_use_system_libzip_libpcre.patch'
 *   environment, line 1086:  Called die
 * The specific snippet of code:
 *               die "Failed Patch: ${patchname}!";



PATCH COMMAND:  patch -p0 -g0 -E --no-backup-if-mismatch  < '/var/lib/layman/hhvm/dev-php/hhvm/files/hhvm_use_system_libzip_libpcre.patch'

================================================
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- a/third-party/CMakeLists.txt
|+++ b/third-party/CMakeLists.txt
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored

**Comment by reanimus** [comment]
@lutel thanks for pointing this out, it should be fixed by my latest commit. I'm building it now to confirm.

For future reference, it'd probably be easier if you filed a bug at the repo itself for things.

**Comment by javer** [comment]
This has been fixed in newer versions of HHVM 3.3.3 and 3.5.0.
